sound/soc/intel/boards/sof_sdw_max98373.c

```diff
@@ -140,9 +155,17 @@ int sof_sdw_mx8373_init(struct snd_soc_card *card,
 	if (info->amp_num == 2)
 		dai_links->init = spk_init;
 
-	info->codec_card_late_probe = mx8373_sdw_late_probe;
-
-	dai_links->ops = &max_98373_sdw_ops;
+	maxim_part_id = info->part_id;
+	switch (maxim_part_id) {
+	case SOF_SDW_PART_ID_MAX98363:
+		dai_links->ops = &max_98363_sdw_ops;
+		break;
+	default:
+		/* Default is considered as MAX98373 soundwire */
+		info->codec_card_late_probe = mx8373_sdw_late_probe;
+		dai_links->ops = &max_98373_sdw_ops;
+		break;
+	}
```

Comment:
this is rather ugly. Can we try to make the code more common but avoid sticking 98363 support sdw_sof_max98373.c?

Reply:
Renamed file to sdw_sof_maxim and internal functions are renamed accordingly